core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -421,6 +423,7 @@ public final void runInContext() {
     @Override
     public void closed(Status status, Metadata trailers) {
       Long timeoutMicros = getRemainingTimeoutMicros(callOptions.getDeadlineNanoTime());
+      transportFuture.cancel(false);
```

Comment:
> If this is false, and the transport future is already being created, the future will complete. false just makes it so if it has not yet been started, it will never start.

Independent of `mayInterruptIfRunning`, `cancel()` causes the future to be cancelled _immediately_ unless it already is complete or failed. Also independent of `mayInterruptIfRunning`, `cancel()` should prevent any processing if processing has not started (which doesn't really mean much for `SettableFuture`s). The only difference with `mayInterruptIfRunning` is whether a currently-running task is attempted to be aborted (e.g., whether `Thread.interrupt()` may be called).

So in practice, the value of `mayInterruptIfRunning` doesn't make much difference, and `false` is generally is a good default.